third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs

```diff
@@ -0,0 +1,594 @@
+        let type_parameters = self.signature(ctx, loc, inst);
+        let cache_key = (struct_env.get_qualified_id(), type_parameters);
+        if let Some(idx) = self.struct_def_inst_to_idx.get(&cache_key) {
+            return *idx;
+        }
+        let def = self.struct_def_index(ctx, loc, struct_env);
+        let struct_inst = FF::StructDefInstantiation {
+            def,
+            type_parameters,
+        };
+        let idx = FF::StructDefInstantiationIndex(ctx.checked_bound(
+            loc,
+            self.module.struct_def_instantiations.len(),
+            MAX_STRUCT_DEF_INST_COUNT,
+            "struct instantiation",
+        ));
+        self.module.struct_def_instantiations.push(struct_inst);
+        self.struct_def_inst_to_idx.insert(cache_key, idx);
+        idx
+    }
+
+    /// Obtains or creates a field handle index.
+    pub fn field_index(
+        &mut self,
+        ctx: &ModuleContext,
+        loc: &Loc,
+        field_env: &FieldEnv,
+    ) -> FF::FieldHandleIndex {
+        let key = (
+            field_env.struct_env.get_qualified_id(),
+            field_env.get_offset(),
+        );
+        if let Some(idx) = self.field_to_idx.get(&key) {
+            return *idx;
+        }
+        let field_idx = FF::FieldHandleIndex(ctx.checked_bound(
+            loc,
+            self.module.field_handles.len(),
+            MAX_FIELD_COUNT,
+            "field",
+        ));
+        let owner = self.struct_def_index(ctx, loc, &field_env.struct_env);
+        self.module.field_handles.push(FF::FieldHandle {
+            owner,
+            field: field_env.get_offset() as FF::MemberCount,
+        });
+        self.field_to_idx.insert(key, field_idx);
+        field_idx
+    }
+
+    /// Obtains or creates a field instantiation handle index.
+    pub fn field_inst_index(
+        &mut self,
+        ctx: &ModuleContext,
+        loc: &Loc,
+        field_env: &FieldEnv,
+        inst: Vec<Type>,
+    ) -> FF::FieldInstantiationIndex {
+        let type_parameters = self.signature(ctx, loc, inst);
+        let key = (
+            field_env.struct_env.get_qualified_id(),
+            field_env.get_offset(),
+            type_parameters,
+        );
+        if let Some(idx) = self.field_inst_to_idx.get(&key) {
+            return *idx;
+        }
+        let field_inst_idx = FF::FieldInstantiationIndex(ctx.checked_bound(
+            loc,
+            self.module.field_instantiations.len(),
+            MAX_FIELD_INST_COUNT,
+            "field instantiation",
+        ));
+        let handle = self.field_index(ctx, loc, field_env);
+        self.module
+            .field_instantiations
+            .push(FF::FieldInstantiation {
+                handle,
+                type_parameters,
+            });
+        self.field_inst_to_idx.insert(key, field_inst_idx);
+        field_inst_idx
+    }
+
+    /// Obtains or generates a constant index.
+    pub fn cons_index(
```

Comment:
This would be much better named `const_index`.   The name you have makes me think of Lisp.

Reply:
Renamed